src/primitives/polynomial.rs

```diff
@@ -0,0 +1,281 @@
+// SPDX-License-Identifier: MIT
+
+//! Polynomials over Finite Fields
+
+#[cfg(all(feature = "alloc", not(feature = "std"), not(test)))]
+use alloc::vec::Vec;
+use core::{iter, ops};
+
+use super::Field;
+
+/// A polynomial over some field.
+#[derive(PartialEq, Eq, Clone, Debug, Hash)]
+pub struct Polynomial<F> {
+    inner: Vec<F>,
+}
+
+impl<F> Polynomial<F> {
+    /// Constructor for a polynomial from a vector of coefficients.
+    ///
+    /// These coefficients are in "little endian" order. That is, the ith
+    /// coefficient is the one multiplied by x^i.
+    pub fn new(f: Vec<F>) -> Self { Self { inner: f } }
+}
+
+impl<F: Field> Polynomial<F> {
+    /// The degree of the polynomial.
+    ///
+    /// For constants it will return zero, including for the constant zero.
+    pub fn degree(&self) -> usize {
+        let mut deg = self.inner.len() - 1;
+        while deg > 0 && self.inner[self.inner.len() - 1 - deg] == F::ZERO {
+            deg -= 1;
+        }
+        deg
```

Comment:
What do you think about using iterators here?

```rust
self.inner.iter().rev().take_while(|el| el == Fe::ZERO).count()
```

Reply:
I like it

Reply:
Actually, after finangling this enough to be correct, it becomes
```rust
self.inner.len() + 1 - self.inner.iter().rev().take_while(|el| **el == F::ZERO).count()
```
which isn't obviously better.